### PR TITLE
Reconnect subscriptions on reconnect

### PR DESCRIPTION
**Pull request by epoplavskis**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-protobuf>=3.2.0
+protobuf>=3.5.1
```
[the remark this answers is not in the document]
Comment (epoplavskis) [reply requirements.txt]: the old version has some issue with byte stream conversion, so just locked down the version to one which doens't have it anymore.

File: photonpump/connection.py [hunk 2]
```diff
@@ -722,7 +740,7 @@ def close(self):
         cmd = convo.ConnectPersistentSubscription(
             subscription, stream, credentials=self.credential
         )
-        future = await self.protocol.enqueue_conversation(cmd)
+        future = await self.protocol.enqueue_conversation(cmd, True)
```
Comment (jlee-made) [inline photonpump/connection.py]: Seems like there should be a test covering this code?

[the remark this answers is not in the document]
Comment (epoplavskis) [reply photonpump/connection.py]: well, when you hit connect_subscription it's going to fail connection, but then retry once eventstore connection is there.

Comment (epoplavskis) [reply photonpump/connection.py]: added a test for this.

File: requirements.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlee-made) [reply requirements.txt]: :+1:

File: photonpump/connection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlee-made) [reply photonpump/connection.py]: I'm not sure about that, but looking at the new test you added, the client code usually uses module-level function `connect` with an async with statement, which is going to `.connect` the `Connection`, so by the time there is the opportunity to `.connect_subscription`, you're already connected to event store.

**Pull request merged**